src/voters/voterFile/voterFile.controller.ts

```diff
@@ -127,4 +132,58 @@ export class VoterFileController {
   ) {
     return this.voterFileFilterService.create(campaign.id, voterFileFilter)
   }
+
+  @Get('filters')
+  @UseCampaign()
+  listVoterFileFilters(@ReqCampaign() campaign: Campaign) {
+    return this.voterFileFilterService.findByCampaignId(campaign.id)
+  }
+
+  @Get('filter/:id')
+  @UseCampaign()
+  async getVoterFileFilter(
+    @Param('id', ParseIntPipe) id: number,
+    @ReqCampaign() campaign: Campaign,
+  ) {
+    const filter: VoterFileFilter | null =
+      await this.voterFileFilterService.findByIdAndCampaignId(id, campaign.id)
+    if (!filter) {
+      throw new NotFoundException('Voter file filter not found')
+    }
+    return filter
+  }
+
+  @Put('filter/:id')
+  @UseCampaign()
+  async updateVoterFileFilter(
+    @Param('id', ParseIntPipe) id: number,
+    @Body() body: UpdateVoterFileFilterSchema,
+    @ReqCampaign() campaign: Campaign,
+  ) {
+    const filter: VoterFileFilter | null =
+      await this.voterFileFilterService.findByIdAndCampaignId(id, campaign.id)
+    if (!filter) {
+      throw new NotFoundException('Voter file filter not found')
+    }
+    return this.voterFileFilterService.updateByIdAndCampaignId(
+      id,
+      campaign.id,
+      body,
+    )
+  }
+
+  @Delete('filter/:id')
+  @UseCampaign()
+  @HttpCode(HttpStatus.NO_CONTENT)
+  async deleteVoterFileFilter(
+    @Param('id', ParseIntPipe) id: number,
+    @ReqCampaign() campaign: Campaign,
+  ) {
+    const filter: VoterFileFilter | null =
+      await this.voterFileFilterService.findByIdAndCampaignId(id, campaign.id)
+    if (!filter) {
+      throw new NotFoundException('Voter file filter not found')
```

Comment:
We shouldn't need this check for delete operations.. deletes can just return no content regardless of whether the operation was successful or not